From 10e2cabbd4b26d33e6d00cf312650180857350f0 Mon Sep 17 00:00:00 2001
From: Xiaozhe Shi <xiaozhes@codeaurora.org>
Date: Wed, 6 May 2015 16:09:49 -0700
Subject: [PATCH 294/959] power: qpnp-smbcharger-mmi: add api to force charger
 tlim

Add an API through the power supply property
POWER_SUPPLY_PROP_FORCE_TLIM in order to force on the charger therm
circuitry.

CRs-Fixed: 793985
Change-Id: Iddcd069e0075c4863247d75bde675e3b03310cf4
Signed-off-by: Xiaozhe Shi <xiaozhes@codeaurora.org>
---
 drivers/power/qpnp-smbcharger-mmi.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index 1506bfbb2e6..9529e41a14c 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -2992,6 +2992,23 @@ static int smbchg_otg_pulse_skip_enable(struct smbchg_chip *chip, bool enable)
 	return 0;
 }
 
+#define LOW_PWR_OPTIONS_REG	0xFF
+#define FORCE_TLIM_BIT		BIT(4)
+static int smbchg_force_tlim_en(struct smbchg_chip *chip, bool enable)
+{
+	int rc;
+
+	rc = smbchg_sec_masked_write(chip, chip->otg_base + LOW_PWR_OPTIONS_REG,
+			FORCE_TLIM_BIT, enable ? FORCE_TLIM_BIT : 0);
+	if (rc < 0) {
+		dev_err(chip->dev,
+			"Couldn't %s otg force tlim rc = %d\n",
+			enable ? "enable" : "disable", rc);
+		return rc;
+	}
+	return rc;
+}
+
 static int smbchg_battery_set_property(struct power_supply *psy,
 				       enum power_supply_property prop,
 				       const union power_supply_propval *val)
@@ -3051,6 +3068,9 @@ static int smbchg_battery_set_property(struct power_supply *psy,
 	case POWER_SUPPLY_PROP_FLASH_ACTIVE:
 		rc = smbchg_otg_pulse_skip_enable(chip, val->intval);
 		break;
+	case POWER_SUPPLY_PROP_FORCE_TLIM:
+		rc = smbchg_force_tlim_en(chip, val->intval);
+		break;
 	default:
 		return -EINVAL;
 	}
-- 
2.11.0

